database/migrations/default/1742408937781_crash_ems_link/up.sql

```diff
@@ -0,0 +1,391 @@
+                crash_pk = NULL,
+                matched_crash_pks = matched_crash_ids
+            WHERE id = matching_ems.id;
+        END IF;
+    END LOOP;
+
+    IF TG_OP = 'UPDATE' THEN
+        UPDATE ems__incidents
+        SET crash_pk = NULL,
+            crash_match_status = 'unmatched',
+            matched_crash_pks = NULL
+        WHERE crash_pk = NEW.id
+          AND (
+              incident_received_datetime  < (NEW.crash_timestamp - time_threshold)
+              OR incident_received_datetime  > (NEW.crash_timestamp + time_threshold)
+              OR geometry IS NULL
+              OR NEW.position IS NULL
+              OR (NOT ST_DWithin(geometry::geography, NEW.position::geography, meters_threshold))
+          )
+          AND 
+            -- ignore manual qa matches: these should not be modified
+            crash_match_status != 'matched_by_manual_qa';
+    END IF;
+    RETURN NEW;
+END;
+$$ language plpgsql;
+
+
+--
+-- create insert trigger
+--
+create or replace trigger crash_insert_ems_match_trigger
+after insert on crashes
+for each row execute function update_crash_ems_match();
+
+--
+-- create update trigger
+--
+create or replace trigger crash_update_ems_match_trigger
+after update on crashes
+for each row when (
+    (new.latitude is distinct from old.latitude)
+    or (new.longitude is distinct from old.longitude)
+    or (new.crash_timestamp is distinct from old.crash_timestamp)
+) execute function update_crash_ems_match();
+
+
+--
+-- remove apd_incident_number parsing from this trigger
+--
+CREATE OR REPLACE FUNCTION public.ems_incidents_trigger()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+BEGIN
+  NEW.geometry = ST_SetSRID(ST_MakePoint(NEW.longitude, NEW.latitude), 4326);
+
+  SELECT 
+    COALESCE(EXISTS (
+      SELECT 1 
+      FROM geo.jurisdictions 
+      WHERE jurisdiction_label = 'AUSTIN FULL PURPOSE'
+      AND ST_Contains(geometry, NEW.geometry)
+    ), FALSE),
+    locations.location_id
+  INTO 
+    NEW.austin_full_purpose,
+    NEW.location_id
+  FROM (SELECT 1) AS dummy
+  LEFT JOIN atd_txdot_locations locations ON (
+    locations.location_group = 1 
+    AND NEW.geometry && locations.geometry 
+    AND ST_Contains(locations.geometry, NEW.geometry)
+  );
+  RETURN NEW;
+END;
+$function$;
+
+-- 
+-- remove incident numbers from trigger condition
+--
+drop trigger if exists ems_incidents_trigger_update on ems__incidents;
+
+create trigger ems_incidents_trigger_update before update on ems__incidents for each row
+    when (old.latitude is distinct from new.latitude or old.longitude is distinct from new.longitude)
+    execute function ems_incidents_trigger();
+
+--
+-- drop these individual incident number columns. a record may contain dozens of apd_incident_numbers
+-- 
+alter table ems__incidents drop column if exists apd_incident_number_1;
+alter table ems__incidents drop column if exists apd_incident_number_2;
+
+--
```

Comment:
we had an unused `ems_id` column on the `people` table, but i think it's best to keep this on the `ems__incidents` just to keep things organized.